auth/libraries/pip-auth/authlib/validator.py

```diff
@@ -0,0 +1,80 @@
+import jwt
+import requests
+import secrets
+from Crypto.Hash import SHA512
+from urllib import parse
+from typing import Optional, Tuple, Dict, Any, Callable
+import simplejson as json
+from .http_handler import HttpHandler
+from .exceptions import InvalidCredentialsException, AuthenticateException
+
+from authlib.config import get_private_token_key, get_private_cookie_key
+
+
+AUTHENTICATION_HEADER = "Authentication"
+REFRESH_ID = "refreshId"
+USER_ID = "userId"
+
+ANONYMOUS_USER = 0
+INVALID_USER = -1
+
+
+class Validator:
+    """
+    The validator verifies a given ticket if it is valid.
+    """
+
+    auth_url = ""
+    debug_fn = print
+    http_handler = None
+
+    def __init__(self, http_handler: HttpHandler, debug_fn: Any = print) -> None:
+        self.http_handler = http_handler
+        self.debug_fn = debug_fn
+
+    def verify_ticket(self, token_encoded: str, cookie_encoded: str) -> int:
+        if not isinstance(token_encoded, str) or not isinstance(cookie_encoded, str):
+            return ANONYMOUS_USER
+        if not self.is_bearer(token_encoded) or not self.is_bearer(cookie_encoded):
```

Comment:
Enhancement: Split the check to get a better error message: `Wrong format of cookie` or `Wrong format of token`